dask_expr/io/tests/test_delayed.py

```diff
@@ -46,3 +47,14 @@ def test_from_delayed_dask():
         assert_eq(s, df.a)
         assert list(s.map_partitions(my_len).compute()) == [1, 2, 3, 4]
         assert ddf.known_divisions == (divisions is not None)
+
+
+@dask.delayed
+def _load(x):
+    return pd.DataFrame({"x": x, "y": [1, 2, 3]})
+
+
+def test_from_delayed_fusion():
+    df = from_delayed([_load(x) for x in range(10)], meta={"x": "int64", "y": "int64"})
+    result = df.map_partitions(lambda x: None, meta={}).optimize().dask
+    assert len(result) == 30
```

Comment:
can you help me out...? `10 x load + 10 x lambda = 20`. What am I missing?

Reply:
if you want to test that it doesn't fuse, maybe

`df.map_partitions(lambda x: None, meta={}).optimize(fuse=False).dask == df.map_partitions(lambda x: None, meta={}).optimize().dask`

is a better test. or even

```
ddf = df.map_partitions(lambda x: None, meta={})
dsk_opt = ddf.optimize().dask
dsk_raw = ddf.lower_completely().dask
```

Reply:
10 x an intermediate task representing the from_delayed thing that isn't fused anymore. Previously that was fused into the map_partitions thing as well. This is the tradeoff for now

Reply:
I like that, adjusted